tls/s2n_kem_preferences.h

```diff
@@ -28,14 +28,26 @@ struct s2n_kem_preferences {
     /* tls13_kem_groups used for hybrid TLS 1.3 */
     uint8_t tls13_kem_group_count;
     const struct s2n_kem_group **tls13_kem_groups;
+
+    /* Which draft revision data format should the client use in its ClientHello. Currently the server will auto-detect
+     * the format the client used from the TotalLength, and will match the client's behavior for backwards compatibility.
+     *
+     * Link: https://datatracker.ietf.org/doc/html/draft-ietf-tls-hybrid-design
+     *  - Draft 0:   PQ Hybrid KEM format: (Total Length, PQ Length, PQ Share, ECC Length, ECC Share)
+     *  - Draft 1-5: PQ Hybrid KEM format: (Total Length, PQ Share, ECC Share)
```

Comment:
Yeah, since the PQ KeyShares are so large compared to ECDHE Shares, a lot of effort has been put into finding ways to shave off any "unnecessary" bytes for PQ KeyShares, even if it differs from how things have historically been encoded. It's theoretically memory safe (since the length of the PQ and ECC Shares should be able to be deterministically looked up by the KEM Algorithm ID) but it requires slightly more complex logic to handle, which can introduce it's own set of bugs.